lib/px/utils/pxclient.lua

```diff
@@ -219,6 +219,22 @@ function M.load(config_file)
         _M.forward_to_perimeterx(px_config.client_host, px_config.client_port_overide)
     end
 
+    function _M.reverse_px_captcha()
+        if not px_config.first_party_enabled then
+            ngx.header["Content-Type"] = 'application/javascript';
+            ngx.say('')
+            ngx.exit(ngx.OK)
+            return
+        end
+        local reverse_prefix = string.sub(px_config.px_appId, 3, string.len(px_config.px_appId))
+        local px_request_uri = string.gsub(ngx.var.uri, '/' .. reverse_prefix .. px_constants.FIRST_PARTY_CAPTCHA_PATH, '')
+        px_logger.debug("Forwarding request from "  .. ngx.var.request_uri .. " to px captcha at " .. px_config.captcha_script_host .. px_request_uri)
+        ngx_req_set_uri(px_request_uri)
+
+        px_common_utils.clear_first_party_sensitive_headers(px_config.sensitive_headers);
+        _M.forward_to_perimeterx(px_config.captcha_script_host)
```

Comment:
need to add here port overide so we could pass a different port, not for clients use usually but its better that we have this under control

Reply:
I disagree. It's fine that `forward_to_perimeterx` accepts an override port because it makes it more generic, but for this flow I don't think there's a reason it would be configurable. we should also consider that a large number of config values is confusing for customers and potential developers